### DISCUSSION TITLE
config timeout with CURLOPT_TIMEOUT_MS instead of CURLOPT_TIMEOUT for curl implementation

### DISCUSSION THREAD

**Discussion by wxl374** [open-telemetry/opentelemetry-cpp#1254] [General]
When configure OtlpHttpExporter's timeout less than 1s, it will set to 0 for the use of CURLOPT_TIMEOUT(https://github.com/open-telemetry/opentelemetry-cpp/blob/fd5a6218bde5a728019f377a67f75c7b821f76c7/ext/include/opentelemetry/ext/http/client/curl/http_operation_curl.h#L189), then never times out (sometimes limited by CURLOPT_LOW_SPEED_TIME). Why not use CURLOPT_TIMEOUT_MS instead？

**Comment by lalitb** [comment]
This is a good point. Would be good to see the libcurl version having CURLOPT_TIMEOUT_MS  before supporting this. Moving this to issue to investigate further.
